src/main/java/com/synopsys/integration/alert/web/user/UserActions.java

```diff
@@ -98,61 +81,53 @@ public UserConfig createUser(UserConfig userConfig) throws AlertDatabaseConstrai
         return convertToCustomUserRoleModel(userModel);
     }
 
-    public UserConfig updateUser(UserConfig userConfig) throws AlertFieldException {
-        //TODO when the userId is in the user config look up the user by id and update the username.
-        if (StringUtils.isNotBlank(userConfig.getPassword())) {
-            updatePassword(userConfig.getUsername(), userConfig.getPassword());
-        }
-        updateEmail(userConfig.getUsername(), userConfig.getEmailAddress());
+    public UserConfig updateUser(UserConfig userConfig) throws AlertFieldException, AlertDatabaseConstraintException {
+        Long userId = Long.valueOf(userConfig.getId());
+        Optional<UserModel> userModel = userAccessor.getUser(userId);
+        if (userModel.isPresent()) {
+            UserModel existingUser = userModel.get();
+            String userName = userConfig.getUsername();
+            String password = userConfig.getPassword();
+            String emailAddress = userConfig.getEmailAddress();
+            boolean passwordHasValue = StringUtils.isNotBlank(password);
+            Map<String, String> fieldErrors = new HashMap<>();
+            validateUserName(fieldErrors, userName);
+            validateRequiredField(FIELD_KEY_USER_MGMT_EMAILADDRESS, fieldErrors, emailAddress);
+            if (!userConfig.isPasswordSet() || passwordHasValue) {
+                validatePasswordLength(fieldErrors, password);
+            }
+            if (!fieldErrors.isEmpty()) {
+                throw new AlertFieldException(fieldErrors);
+            }
 
-        Optional<UserModel> userModel = userAccessor.getUser(userConfig.getUsername());
-        Set<String> configuredRoleNames = userConfig.getRoleNames();
-        if (null != configuredRoleNames && !configuredRoleNames.isEmpty() && userModel.isPresent()) {
-            Collection<UserRoleModel> roleNames = authorizationUtility.getRoles().stream()
-                                                      .filter(role -> configuredRoleNames.contains(role.getName()))
-                                                      .collect(Collectors.toList());
-            authorizationUtility.updateUserRoles(userModel.get().getId(), roleNames);
-        }
+            if (passwordHasValue) {
+                UserModel newUserModel = UserModel.existingUser(existingUser.getId(), userName, password, emailAddress, existingUser.getRoles());
+                userAccessor.updateUser(newUserModel, false);
```

Comment:
This object creation and user upating can instead be done outside of the if statement. We can use the if statement to update the password value. The passwordEncoded field will just be equal to negation of passwordHasValue